--- mbedtls-1.3.10/library/pkparse.c    2015-02-10 18:00:42.000000000 +0200
+++ mbedtls-1.3.10p/library/pkparse.c   2015-05-10 18:59:43.911605796 +0300
@@ -1243,6 +1243,17 @@
 
     ret = pk_parse_subpubkey( &p, p + keylen, ctx );
 
+#if defined(POLARSSL_RSA_C)
+    if( ret != 0 )
+    {
+        pk_init_ctx( ctx, pk_info_from_type( POLARSSL_PK_RSA ) );
+        p = (unsigned char *) key;
+        ret = pk_get_rsapubkey( &p, p + keylen, pk_rsa( *ctx ) );
+        if( ret != 0)
+          pk_free( ctx );
+    }
+#endif /* POLARSSL_RSA_C */
+
 #if defined(POLARSSL_PEM_PARSE_C)
     pem_free( &pem );
 #endif
